pkg/mining/miningGroup.go

```diff
@@ -191,15 +197,40 @@ type Result struct {
 }
 
 // dispatches a chunk and returns the number of hashes chosen.
-func (b *Backend) dispatchWork(hash *HashSettings, start uint64, resultCh chan *backendResult) uint64 {
+func (b *Backend) dispatchWork(parentCtx context.Context, hash *HashSettings, start uint64, resultCh chan *backendResult) uint64 {
 	target := b.HashRateEstimate * targetChunkTime.Seconds()
 	step := b.StepSize()
 	nsteps := uint64(math.Round(target / float64(step)))
 	if nsteps == 0 {
 		nsteps = 1
 	}
 	n := nsteps * step
-	go b.doWork(hash, start, n, resultCh)
+
+	var err error
+	var timeOfLastNewValue *big.Int
+	var retriesCount int
+	for {
+		// checks the last submit value in the oracle and set a timeout of 15min - (now-lastSubmit).
+		// This is because 15min after the last submit any solution will work.
+		timeOfLastNewValue, err = b.contractInstance.GetUintVar(nil, rpc.Keccak256([]byte("_TIME_OF_LAST_NEW_VALUE")))
+		if err == nil {
+			break
+		}
+		retriesCount++
+		level.Error(b.logger).Log("msg", "getting time of last new value from oracle, retrying in a second", "err", err, "numRetry", retriesCount)
+		if retriesCount == 10 {
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	if retriesCount == 10 {
+		// TODO: do work and return any solution?
+		return 0
+	}
```

Comment:
If getting the last submit time returned and error there is no point to continue as most likely there is something wrong with the node so no solution could be submitted anyway. Lets keep the things simple and just return an error when can't get the time of last submit.
```suggestion
	if retriesCount == 10 {
		// TODO: do work and return any solution?
		return 0
	}
```